srcs/executing/exec_cmd_system.c

```diff
@@ -0,0 +1,108 @@
+#include "minishell.h"
+
+/*
+** If command is a relative path to an executable
+** that path is stored as the command path.
+** 
+** Checks if the command exists on each path and is executable
+** Access function: F_OK - command exists
+**                  X_OK - is executable
+** If yes, the path is stored in that cmd_table, ready to be used.
+*/
+int	valid_command(t_minishell *minishell, t_cmd_table *cmd)
+{
+	int	i;
+
+	if (access(cmd->cmd_name, F_OK) == 0
+			&& access(cmd->cmd_name, X_OK) == 0)
+	{
+		cmd->cmd_path = ft_strdup(cmd->cmd_name);
+		return (1);
+	}
+	get_cmd_paths(minishell);
+	i = 0;
+	while (minishell->env_paths[i])
+	{
+		if (cmd->cmd_path)
+			free(cmd->cmd_path);
+		cmd->cmd_path = ft_strjoin(minishell->env_paths[i], cmd->cmd_name);
+		if (access(cmd->cmd_path, F_OK) == 0
+			&& access(cmd->cmd_path, X_OK) == 0)
+			return (1);
+		i++;
+	}
+	return (0);
+}
+
+static char **build_execve_args(t_cmd_table *cmd)
```

Comment:
that's exactly why I use them, because there's everything in there, command name, flags and arguments. I need to pass all of that to execve. Of course that I could copy cmd_name, then cmd_flags and then cmd_args, the effect would be the same, but this way it's simpler and shorter. So what this function does, if you run
ls -l -a srcs
it creates the array ["ls", "-l", "-a", "srcs"] to be used as execve, as it needs to receive the information in this format

Reply:
ah, hum. then maybe we need to check that, because if you receive them in the wrong order, does it still work ? (like ls srcs -l -a ?)